h2o-bindings/bin/bindings.py

```diff
@@ -368,15 +368,22 @@ def find_field(fields, field_name):
             # Write the generic information about the base class
             schema = m[base]
             schema["generics"] = generics
-            generic_map = {long_type: gen_type for gen_type, long_type in generics}
+            generic_map = defaultdict(list)
+            for name, schema_name in generics:
+                generic_map[schema_name].append(name)     
             generic_index = {geninfo[0]: i for i, geninfo in enumerate(generics)}
             mapped_fields = {}
             for field in schema["fields"]:
                 ftype = field["schema_name"]
                 if ftype in generic_map:
                     gen_type = generic_map[ftype]
-                    field["schema_name"] = gen_type
-                    mapped_fields[field["name"]] = generic_index[gen_type]
+                    if len(gen_type) > 1:
+                        field["schema_name"] = gen_type[0]
+                        mapped_fields[field["name"]] = generic_index[gen_type[0]]
+                        gen_type.pop(0)
+                    else:    
+                        field["schema_name"] = gen_type[0]
+                        mapped_fields[field["name"]] = generic_index[gen_type[0]]
```

Comment:
looks like only `gen_type.pop(0)` should be in the `if` condition, the other statements are always executed.